include/superior_mysqlpp/types/array.hpp

```diff
@@ -14,18 +14,29 @@ namespace SuperiorMySqlpp
 {
     /**
      * Base class for all array based SQL types
+     *
+     * If TerminatingZero == true, hidden empty byte is always appended after payload.
+     * Mind the fact that storage needs to be (at least one byte) larger for this mode.
      */
-    template<std::size_t N>
+    template<std::size_t N, bool TerminatingZero = false>
     class ArrayBase
```

Comment:
Just tip for the future - maybe it would be worth it to change this type to `template<typename, size_t, bool>` and possibly create an alias e.g. `template<size_t sz, boo; b> using ByteArrayBase = ArrayBase<(unsigned?) char, sz, b>;`.
The reason why I suggest this is simple: alignment and reusability.

Reply:
I agree that this idea has merit, but I'd rather pursue it later, in more mature and robust form that allows users to deserialize binary data to arbitrary types - solving problems like alignment and `std::string` support in one go.
In any case, I feel it's out of scope of this PR.

Reply:
I agree.